README.md

```diff
@@ -0,0 +1,494 @@
+})
+```
+
+### `logger.flush()`
+
+* Emits: [`cleared`](#event-cleared)
+
+When `flush` is called, any messages in the buffer are sent to LogDNA.  It's not necessary to call this manually, although it is useful
+to do so to ensure clean shutdown (see [Best Practices](#best-practices)).  When `log` is called, it automatically starts a timer
+which will call `flush`, but it is idempotent and can be called at any time.
+
+If log lines exist in the current buffer, it is pushed onto a send queue, and a new buffer is created.  The send queue is processed and uploaded to LogDNA.
+
+If no work needs to be done, the `cleared` event is immediately emitted.
+
+
+### `logger.log(statement[, options])`
+
+* `statement` [`<String>`][] | [`<Object>`][] - Text or object of the log entry.  Objects are serialized.
+* `options` [`<String>`][] | [`<Object>`][] - A string representing a [level](#supported-log-levels), or object:
+    * `level` [`<String>`][] - Desired [level](#supported-log-levels) for the current message. **Default:** `logger.level`
+    * `app` [`<String>`][] - App name to use for the current message. **Default:** `logger.app`
+    * `env` [`<String>`][] - Environement name to use for the current message. **Default:** `logger.env`
+    * `timestamp` [`<Number>`][] - Epoch ms time to use for the current message. Must be within 24 hours. **Default:** `Date.now()`
+    * `context` [`<Object>`][] - Synonym for `meta`, but mutually exclusive; Ignored if `meta` exists.
+    * `indexMeta` [`<Boolean>`][] - Allows for the `meta` to be searchable in LogDNA. **Default:** `logger.indexMeta`
+    * `meta` [`<Object>`][] - Per-message meta data.  Combined with key-values created with `addMetaProperty`
+* Emits: [warn](#event-warn), [error](#event-error)
+
+Sends a string or object to LogDNA for storage.  If the [convenience methods](#convenience-methods) are used, they call this function
+under the hood, so the options are the same.  The only difference is that `level` is automatically set in the convenience methods.
+
+
+### `logger.removeMetaProperty(key)`
+
+* `key` [`<String>`][] - The key (and implied value) to be remove from the global `meta` object.
+* Emits: [`warn`](#event-warn)
+
+Attempts to remove the given key from the global `meta` data object.  If the key is not found, `warn` is emitted.
+
+
+## How Log Lines are Sent to LogDNA
+
+In default operation, when `log` functions are called, the line is added to a buffer to cut down on HTTP traffic to the server.  
+The buffer is flushed every `flushIntervalMs` milliseconds, or if the log line lengths grow beyond `flushLimit`.
+
+When `flush` fires (or is called manually), the current buffer is put onto a send queue, and a new buffer is started.  The send queue begins
+sending to LogDNA.  It will continue to send without pausing or honoring `flushIntervalMs` as long as there are buffers in the send queue.  
+When the send queue is empty, `cleared` is emitted.
```

Comment:
```suggestion
When `flush` fires (or is called manually), the current buffer is put onto a send queue, and a new buffer is started. The send queue begins
sending to LogDNA. It will continue to send without pausing or honoring `flushIntervalMs` as long as there are buffers in the send queue.  
When the send queue is empty, `cleared` is emitted.
```